core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -307,12 +336,17 @@ public void onNext(ByteBuffer byteBuffer) {
             if (byteBuffer == null) {
                 throw new NullPointerException("onNext must not be called with null byteBuffer");
             }
+            if (isCancelled.get()) {
```

Comment:
Is this needed? If publisherToUpstream.error is invoked, send will be no op

Reply:
Did some testing around this, yep I think  you are right, this is redundant. I will remove it.